code/datums/helper_datums/map_template.dm

```diff
@@ -64,4 +65,50 @@
 	var/list/filelist = flist(path)
 	for(var/map in filelist)
 		var/datum/map_template/T = new(path = "[path][map]", rename = "[map]")
-		map_templates[T.name] = T
\ No newline at end of file
+		map_templates[T.name] = T
+
+	if(!config.disable_space_ruins) // so we don't unnecessarily clutter start-up
+		preloadRuinTemplates()
+	//preloadShuttleTemplates()
+
+/proc/preloadRuinTemplates()
+	// Still supporting bans by filename
+	var/list/banned
+	if(fexists("config/spaceRuinBlacklist.txt"))
+		banned = generateMapList("config/spaceRuinBlacklist.txt")
+	else
+		banned = generateMapList("config/example/spaceRuinBlacklist.txt")
+	//banned += generateMapList("config/lavaRuinBlacklist.txt")
+
+	for(var/item in subtypesof(/datum/map_template/ruin))
+		var/datum/map_template/ruin/ruin_type = item
+		// screen out the abstract subtypes
+		if(!initial(ruin_type.id))
+			continue
+		var/datum/map_template/ruin/R = new ruin_type()
+
+		if(banned.Find(R.mappath))
+			continue
+
+		map_templates[R.name] = R
+		ruins_templates[R.name] = R
+
+		/*
+		if(istype(R, /datum/map_template/ruin/lavaland))
+			lava_ruins_templates[R.name] = R
+		*/
+		if(istype(R, /datum/map_template/ruin/space))
+			space_ruins_templates[R.name] = R
+
+/*
+/proc/preloadShuttleTemplates()
```

Comment:
This will probably be used in the future, I wouldn't really delete it if it isn't totally necessary.